Rubberduck.Core/UI/CodeExplorer/Commands/RemoveCommand.cs

```diff
@@ -31,33 +45,59 @@ protected override void OnExecute(object parameter)
                 return;
             }
 
-            var qualifiedModuleName = node.Declaration.QualifiedName.QualifiedModuleName;
-            var message = string.Format(CodeExplorerUI.ExportBeforeRemove_Prompt, node.Name);
+            RemoveComponent(node.Declaration.QualifiedModuleName);
+        }
 
-            switch (MessageBox.Confirm(message, CodeExplorerUI.ExportBeforeRemove_Caption, ConfirmationOutcome.Yes))
+        public bool RemoveComponent(QualifiedModuleName qualifiedModuleName, bool promptToExport = true)
+        {
+            if (promptToExport && !TryExport(qualifiedModuleName))
             {
-                case ConfirmationOutcome.Yes:
-                    if (!PromptFileNameAndExport(qualifiedModuleName))
-                    {
-                        return;
-                    }
-                    break;
-                case ConfirmationOutcome.No:
-                    break;
-                case ConfirmationOutcome.Cancel:
-                    return;
-                }
-
+                return false;
+            }
+            
             // No file export or file successfully exported--now remove it
-            var components = ProjectsProvider.ComponentsCollection(qualifiedModuleName.ProjectId);
             try
             {
-                components?.Remove(ProjectsProvider.Component(qualifiedModuleName));
+                _projectsRepository.RemoveComponent(qualifiedModuleName);
             }
             catch (Exception ex)
             {
-                MessageBox.NotifyWarn(ex.Message, string.Format(CodeExplorerUI.RemoveError_Caption, qualifiedModuleName.ComponentName));
+                _messageBox.NotifyWarn(ex.Message, string.Format(CodeExplorerUI.RemoveError_Caption, qualifiedModuleName.ComponentName));
+                return false;
+            }
+
+            return true;
+        }
+
+        private bool TryExport(QualifiedModuleName qualifiedModuleName)
+        {
+            var projectId = qualifiedModuleName.ProjectId;
+            var projectType = _projectsRepository.Project(projectId).Type;
```

Comment:
This is missing a null check. If the ID is no longer in the repository, the provider returns `null`.

Reply:
No longer needed - see next comment.